### PR TITLE
feat: Add package version to Nargo.toml metadata

### PR DESCRIPTION
**Pull request by kevaundray**
# Description

This adds the `package_version` field to Nargo.toml with semver validation checks

## Problem\*

Resolves 

## Documentation\*

Check one:
- [ ] No documentation needed.
- [ ] Documentation included in this PR.
- [ ] **[Exceptional Case]** Documentation to be submitted in a separate PR.

# PR Checklist\*

- [ ] I have tested the changes locally.
- [ ] I have formatted the changes with [Prettier](https://prettier.io/) and/or `cargo fmt` on default settings.

### CODE REVIEW

**Review by TomAFrench — COMMENTED** [review]
Why not just `version`? That's fairly standard across `package.json`, `Cargo.toml`, etc. so I don't think that people will be confused on its purpose.

**Comment by kevaundray** [comment]
> Why not just `version`? That's fairly standard across `package.json`, `Cargo.toml`, etc. so I don't think that people will be confused on its purpose.

I didn't do that inititially because we also have `compiler_version`, so it seemed better to be explicit about what we meant when we said version

**Review by vezenovm — COMMENTED** [review]
Can you update the init command in nargo to include a package version or else projects created with nargo new with error out

**Comment by kevaundray** [comment]
> Can you update the init command in nargo to include a package version or else projects created with nargo new with error out

I think that if package_version is not included, then it will do anything. Were you seeing a different behaviour?

**Review by TomAFrench — APPROVED** [review]
I'm still of the opinion that we can have this just be `version`. In rust we have a `version` field which is for the package and a `rust-version` field which specifies the MSRV, this maps directly onto the Nargo behaviour so I don't think it would be confusing. 

https://doc.rust-lang.org/cargo/reference/manifest.html

File: test_programs/execution_success/1_mul/Nargo.toml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 [package]
 name = "1_mul"
+version = "0.1.0"
 type = "bin"
 authors = [""]
 
+
+
```
Comment (TomAFrench) [inline test_programs/execution_success/1_mul/Nargo.toml:8]:
```suggestion
```

**Pull request merged**